pytorch_lightning/trainer/trainer.py

```diff
@@ -2056,10 +2056,12 @@ def devices(self) -> int:
         return self.num_devices
 
     @property
-    def data_parallel_device_ids(self) -> Optional[List[int]]:
-        return (
-            self._accelerator_connector.parallel_device_ids if self._accelerator_connector.parallel_device_ids else None
+    def data_parallel_device_ids(self) -> List[int]:
+        rank_zero_deprecation(
+            "`Trainer.data_parallel_device_ids` was deprecated in v1.6 and will be removed in v1.8."
+            " Please use `Trainer.device_ids` instead."
         )
+        return self.device_ids if isinstance(self.accelerator, GPUAccelerator) else []
```

Comment:
```suggestion
        return self.device_ids if isinstance(self.accelerator, GPUAccelerator) else None
```
a deprecation message only makes sense if we don't change the behavior.

Reply:
> why doesn't this always return self.device_ids ? why is the GPU accelerator check needed?

I think we simply want to keep the behavior during the deprecation phase so everybody who relies on it can keep using it.

Reply:
Make sense. I've changed non-GPU case to keep returning None.

Reply:
thanks